python/mpt/src/mpt/__init__.py

```diff
@@ -0,0 +1,1372 @@
+                    trie_set(
+                        _storage_tries[address],
+                        key,
+                        U256(int.from_bytes(rlp.decode(value), "big")),
+                    )
+
+        return State(
+            _main_trie=_main_trie,
+            _storage_tries=_storage_tries,
+        )
+
+    @classmethod
+    def from_json(cls, path: str) -> "EthereumState":
+        logger.debug(f"Loading MPT from JSON file: {path}")
+        with open(path, "r") as f:
+            data = json.load(f)
+
+        nodes = {
+            keccak256(hex_to_bytes(node)): (hex_to_bytes(node))
+            for node in data["witness"]["state"]
+        }
+
+        codes = {
+            keccak256(hex_to_bytes(code)): hex_to_bytes(code)
+            for code in data["witness"]["codes"]
+        }
+
+        # Process the access list from the JSON data
+        access_list = {
+            Address(hex_to_bytes(item["address"])): (
+                [Bytes32(hex_to_bytes(key)) for key in item["storageKeys"]]
+                if item["storageKeys"] is not None
+                else None
+            )
+            for item in data.get("accessList", [])
+        }
+
+        state_root = Hash32(hex_to_bytes(data["witness"]["ancestors"][0]["stateRoot"]))
+
+        return cls(
+            nodes=nodes,
+            codes=codes,
+            access_list=access_list,
+            state_root=state_root,
+        )
+
+    def get(self, path: Bytes, root_hash: Optional[Hash32] = None) -> Optional[Bytes]:
+        """
+        Get a value from the trie at the given path.
+
+        Parameters
+        ----------
+        path : Bytes
+            The path to look up
+        root_hash : Hash32, optional
+            The root hash to start from, defaults to the trie's state_root
+
+        Returns
+        -------
+        Optional[Bytes]
+            The value at the path, or None if not found
+        """
+        if root_hash is None:
+            root_hash = self.state_root
+
+        logger.debug(f"Getting value for path: {'0x' + path.hex() if path else 'None'}")
+        logger.debug(f"Starting from root hash: 0x{root_hash.hex()}")
+
+        # Check if the root hash exists in our nodes
+        if root_hash not in self.nodes and root_hash != EMPTY_TRIE_ROOT_HASH:
+            logger.error(f"Root hash not found in nodes: {root_hash.hex()}")
+            return None
+
+        # Start traversal from the root
+        nibble_path = bytes_to_nibble_list(path)
+        try:
+            return self._get_node(root_hash, nibble_path)
+        except Exception as e:
+            logger.error(f"Error in get: {str(e)}")
+            return None
+
+    def get_storage_root(self, address: Address) -> Optional[Hash32]:
+        """
+        Get the storage root for an account.
+        """
+        account = self.get(keccak256(address))
+        if account is None:
+            return EMPTY_TRIE_ROOT_HASH
+        decoded = rlp.decode(account)
```

Comment:
(echoing to a previous comment with a class for decoded accounts, we could do this validation when creating the class instance), as I see it's used in a lot of places